lib/openhab/dsl/items/group_item.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'forwardable'
+require 'java'
+require 'openhab/dsl/items/item_command'
+require 'openhab/dsl/items/item_delegate'
+require 'openhab/core/entity_lookup'
+
+module OpenHAB
+  module DSL
+    module Items
+      #
+      # Delegator to OpenHAB Group Item
+      #
+      class GroupItem
+        extend OpenHAB::DSL::Items::ItemCommand
+        extend OpenHAB::DSL::Items::ItemDelegate
+        include Enumerable
+
+        def_item_delegator :@group_item
+
+        #
+        # Indicator struct interpreted by rules to trigger based on items contained in a group
+        #
+        GroupItems = Struct.new(:group, keyword_init: true)
+
+        #
+        # Create a new GroupItem
+        #
+        # @param [Java::Org::openhab::core::items::GroupItem] group_item OpenHAB GroupItem to delegate to
+        #
+        def initialize(group_item)
+          @group_item = group_item
+
+          item_missing_delegate { @group_item }
+          add_state_methods
+          add_command_methods
+        end
+
+        #
+        # Iterates through the direct members of the Group
+        #
+        def each(&block)
+          OpenHAB::Core::EntityLookup.decorate_items(@group_item.members.to_a).each(&block)
```

Comment:
This is not a suggestion, more of a question for me to learn: Why not do item_missing_delegate { @group_item.members } instead of implementing each?